pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -353,16 +361,23 @@ func desiredLoadBalancerService(ci *operatorv1.IngressController, deploymentRef
 		}
 		switch platform.Type {
 		case configv1.AWSPlatformType:
-			if ci.Status.EndpointPublishingStrategy.LoadBalancer != nil &&
-				ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
-				ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.Type == operatorv1.AWSLoadBalancerProvider &&
-				ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil &&
-				ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.Type == operatorv1.AWSNetworkLoadBalancer {
-				service.Annotations[AWSLBTypeAnnotation] = AWSNLBAnnotation
-				// NLBs require a different health check interval than CLBs
-				service.Annotations[awsLBHealthCheckIntervalAnnotation] = awsLBHealthCheckIntervalNLB
-			} else {
-				service.Annotations[awsLBHealthCheckIntervalAnnotation] = awsLBHealthCheckIntervalDefault
+			service.Annotations[awsLBHealthCheckIntervalAnnotation] = awsLBHealthCheckIntervalDefault
+			if lb != nil && lb.ProviderParameters != nil {
+				if aws := lb.ProviderParameters.AWS; aws != nil && lb.ProviderParameters.Type == operatorv1.AWSLoadBalancerProvider {
+					switch aws.Type {
+					case operatorv1.AWSNetworkLoadBalancer:
+						service.Annotations[AWSLBTypeAnnotation] = AWSNLBAnnotation
+						// NLBs require a different health check interval than CLBs.
+						// See <https://bugzilla.redhat.com/show_bug.cgi?id=1908758>.
```

Comment:
Scattering bug numbers around feels like it's going to be a problem once Bugzilla is gone.

Reply:
Many other projects use Bugzilla outside of OpenShift.  I doubt Bugzilla will ever go away (and I would be very disappointed if it did).